### PR TITLE
feat: Added support for default_headers for azure_openai.

### PR DESCRIPTION
**Pull request by a-s-gorski**

### CODE REVIEW

**Comment by a-s-gorski** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by a-s-gorski** [comment]
recheck

**Comment by axiomofjoy** [comment]
I changed the semantic convention to lowercase so CI will pass.

File: tests/experimental/evals/models/test_openai.py [hunk 1]
```diff
@@ -30,6 +30,32 @@ def test_azure_openai_model(monkeypatch):
...
+        header_key = "header"
+        header_value = "my-example-header-value"
+        default_headers = {header_key: header_value}
+        model = OpenAIModel(
+            model="gpt-4-turbo-preview",
+            api_version="2023-07-01-preview",
+            azure_endpoint="https://example-endpoint.openai.azure.com",
+            default_headers=default_headers,
+        )
+        model._init_open_ai()
```
Comment (axiomofjoy) [inline tests/experimental/evals/models/test_openai.py:45]:
```suggestion
```

**Review by axiomofjoy — APPROVED** [review]

**Comment by axiomofjoy** [comment]
@a-s-gorski I went ahead and applied a couple of those comments. We'll get this merged and will cut a new release for you.

**Pull request merged**